src/Css/Style.php

```diff
@@ -1173,18 +1212,103 @@ private function parse_calc(string $expr): array
      * @param float|null $font_size
      * @return float|null
      */
-    private function evaluate_func_calc(array $rpn, float $ref_size = 0, ?float $font_size = null): ?float
+    private function evaluate_func(array $rpn, float $ref_size = 0, ?float $font_size = null): ?float
     {
         if (\count($rpn) === 0) {
             return null;
         }
 
-        $ops = ['*', '/', '+', '-'];
+        $ops = ['*', '/', '+', '-', ','];
 
         $stack = [];
 
         foreach ($rpn as $part) {
-            if (\in_array($part, $ops, true)) {
+            if (\in_array($part, self::CSS_MATH_FUNCTIONS, true)) {
+                $argv = array_pop($stack);
+                if (!is_array($argv)) {
+                    $argv = [$argv];
+                }
+                $argc = \count($argv);
+                switch ($part) {
+                    case 'abs':
+                    case 'acos':
+                    case 'asin':
+                    case 'atan':
+                    case 'cos':
+                    case 'exp':
+                    case 'sin':
+                    case 'sqrt':
+                    case 'tan':
+                        if ($argc !== 1) {
+                            return null;
+                        }
+                        $stack[] = call_user_func_array($part, $argv);
+                        break;
+                    case 'atan2':
+                    case 'hypot':
+                    case 'pow':
+                        if ($argc !== 2) {
+                            return null;
+                        }
+                        $stack[] = call_user_func_array($part, $argv);
+                        break;
+                    case 'log':
+                        if ($argc === 1) {
+                            $stack[] = log($argv[0]);
+                        } elseif ($argc === 2) {
+                            $stack[] = log($argv[0], $argv[1]);
+                        } else {
+                            return null;
+                        }
+                        break;
+                    case 'max':
+                        $stack[] = max($argv);
+                        break;
+                    case 'min':
+                        $stack[] = min($argv);
+                        break;
+                    case 'mod':
+                        if ($argc !== 2) {
+                            return null;
+                        }
+                        $stack[] = $argv[0] % $argv[1];
+                        break;
+                    case 'rem':
+                        if ($argc !== 2) {
+                            return null;
+                        }
+                        $stack[] = fmod($argv[0], $argv[1]);
+                        break;
+                    case 'round':
+                        if ($argc === 2) {
+                            $stack[] = round($argv[0], $argv[1], PHP_ROUND_HALF_UP);
```

Comment:
From first glance, the second number parameter of CSS `round` does not directly map to the precision parameter of PHP `round`. The relation might be something like `precision = -log10(B)`, although PHP `round` only takes integer values for precision. See https://www.w3.org/TR/css-values-4/#round-func.

As with the modulo functions, I would rather leave the function unimplemented instead of incorrectly implemented.

Reply:
I looked into it more closely, it's not math round. Not precision but "step"
I think the correct way is

```
$stack[] = round($argv[0] / $argv[1]) * $argv[1];

["round(0, 20)", null, 0.0],
["round(10, 20)", null, 20.0],
["round(20, 20)", null, 20.0],
["round(25, 20)", null, 20.0],
["round(30, 20)", null, 40.0],
["round(40, 20)", null, 40.0],

["round(0.54, 0.1)", null, 0.5],
["round(0.56, 0.1)", null, 0.6],

```

Reply:
Yes, that’s much simpler. I think you are mostly there now, just need to figure out correct treatment of `A < 0`, and also verify it behaves correctly for `B < 0`. The spec states that the *nearest* rounding strategy rounds in the direction of +∞ if exactly between the bounds. That means `round(-30, 20)` should resolve to `-20`, not `-40` like the current implementation. See https://www.w3.org/TR/css-values-4/#css-round-to-the-nearest-integer.

The other rounding strategies should be similarly straightforward then, since you already have implemented them as part of `mod` and `rem`.